src/PointwiseFunctions/GeneralRelativity/KerrSchildCoords.hpp

```diff
@@ -0,0 +1,115 @@
+ * the spherical coordinates \f$(r, \theta, \phi)\f$ through
+ *
+ * \f{align*}
+ * x &= (r \cos\phi + a \sin\phi)\sin\theta =
+ * \sqrt{r^2 + a^2}\sin\theta\cos\left(\phi - \tan^{-1}(a/ r)\right)\\
+ * y &= (r \sin\phi - a \cos\phi)\sin\theta =
+ * \sqrt{r^2 + a^2}\sin\theta\sin\left(\phi - \tan^{-1}(a/ r)\right)\\
+ * z &= r \cos\theta.
+ * \f}
+ *
+ * Using the identities
+ *
+ * \f{align*}
+ * r^2 &= \frac{x^2 + y^2 + z^2 - a^2}{2} +
+ * \sqrt{\left(\frac{x^2 + y^2 + z^2 - a^2}{2}\right)^2 + a^2 z^2}\\
+ * \sin\theta &= \sqrt{\frac{x^2 + y^2}{r^2 + a^2}},\quad
+ * \cos\theta = \frac{z}{r}\\
+ * \sin\phi &= \frac{x a + y r}{\sqrt{(x^2 + y^2)(r^2 + a^2)}},\quad
+ * \cos\phi = \frac{x r - y a}{\sqrt{(x^2 + y^2)(r^2 + a^2)}},
+ * \f}
+ * the Jacobian is
+ *
+ * \f{align*}
+ * \dfrac{\partial(x, y, z)}{\partial(r, \theta, \phi)} &=
+ * \left(\begin{array}{ccc}
+ * \dfrac{xr - ya}{r^2 + a^2} &
+ * \dfrac{xz}{r}\sqrt{\dfrac{r^2 + a^2}{x^2 + y^2}} & -y\\
+ * \dfrac{xa + yr}{r^2 + a^2} &
+ * \dfrac{yz}{r}\sqrt{\dfrac{r^2 + a^2}{x^2 + y^2}} & x\\
+ * \dfrac{z}{r} & - r\sqrt{\dfrac{x^2 + y^2}{r^2 + a^2}} & 0
+ * \end{array}\right),
+ * \f}
+ *
+ * which can be used to transform tensors between both coordinate systems.
+ */
+class KerrSchildCoords {
+ public:
+  /// The black hole spin magnitude in units of the black hole mass.
+  struct SpinParameter {
+    using type = double;
+    static constexpr OptionString help = {
+        "The dimensionless black hole spin magnitude."};
+  };
+
+  KerrSchildCoords() = default;
+  KerrSchildCoords(const KerrSchildCoords& /*rhs*/) = delete;
+  KerrSchildCoords& operator=(const KerrSchildCoords& /*rhs*/) = delete;
+  KerrSchildCoords(KerrSchildCoords&& /*rhs*/) noexcept = default;
+  KerrSchildCoords& operator=(KerrSchildCoords&& /*rhs*/) noexcept = default;
+  ~KerrSchildCoords() = default;
+
+  explicit KerrSchildCoords(SpinParameter::type spin_parameter) noexcept;
+
+  /// Kerr-SChild \f$r^2\f$ in terms of the Cartesian coordinates.
+  template <typename DataType>
+  Scalar<DataType> r_coord_squared(
+      const tnsr::I<DataType, 3, Frame::Inertial>& cartesian_coords) const
+      noexcept;
+
+  /// The spatial components of the Jacobian of the transformation.
+  template <typename DataType>
+  tnsr::Ij<DataType, 3, Frame::Inertial> jacobian_matrix(
+      const tnsr::I<DataType, 3, Frame::Inertial>& cartesian_coords) const
+      noexcept;
+
+  /// Transforms a spatial vector from Spherical Kerr-Schild coordinates
+  /// to Cartesian Kerr-Schild coordinates.
+  template <typename DataType>
+  tnsr::I<DataType, 3, Frame::Inertial> cartesian_from_spherical(
+      const tnsr::I<DataType, 3, Frame::Inertial>& spatial_vector,
+      const tnsr::I<DataType, 3, Frame::Inertial>& cartesian_coords) const
+      noexcept;
+
+  // clang-tidy: google-runtime-references
+  void pup(PUP::er& /*p*/) noexcept;  // NOLINT
+
+  SpinParameter::type spin_parameter() const noexcept {
```

Comment:
Can you clarify what spin_parameter you mean? dimensionful_spin, dimensionless_spin? Looks like here you mean dimensionless_spin_times_mass? I worry that spin_parameter leaves unclear what spin is being used. 

I also think it might be better to use dimensionless_spin instead, just so it's less likely the wrong spin gets passed in.